lib/drivers/medtronicDriver.js

```diff
@@ -419,7 +419,11 @@ module.exports = function (config) {
 
   function decodeMessage (parser, message) {
     var response = struct.unpack(message, 0, 'b', ['recordType']);
-    _.assign(response, parser(message));
+    var result = parser(message);
+    if(result instanceof Error) {
+      return result;
```

Comment:
OK, I've changed it to `throw` instead and wrapped the `decodeMessage` calls in `try/catch` blocks.